tests/lib/rules/prop-types.js

```diff
@@ -1277,6 +1277,9 @@ ruleTester.run('prop-types', rule, {
         '  return names.map((name) => {',
         '    return <div>{name}</div>;',
         '  });',
+        '}',
+        'Hello.propTypes = {',
+        '  names: React.PropTypes.string',
```

Comment:
Here `Hello` is not a component (it returns an Array, so no `propTypes` should be specified)